enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+> Note: The plain provisioner has the current limitation around applying Bundle contents that will produce runtime issues.
+
+In the case that unpacked Bundle contents contain a Deployment manifest, and that manifest references an invalid container image, then the result would be an unavailable container process after being applied to the cluster. Despite the presence of runtime issues here, the BundleDeployment that manages those manifests would reflect a successful installation state. Addressing this limitation in the context of PO packages is discussed in more detail in subsequent sections.
+
+#### Proxying PO State
+
+```mermaid
+sequenceDiagram
+participant CVO
+participant ClusterOperator
+participant PO Manager
+CVO->>ClusterOperator: Watches
+PO Manager->>ClusterOperator: Manages
+PO Manager-->ClusterOperator: Aggregates PO Package State
+PO Manager->>PO Custom Resources: Watches
+PO Custom Resources->>PO Manager: Relays .Status
+BundleDeployments->>PO Custom Resources: Relays .Status
+```
+
+The PO manager will be responsible for acting as a proxy between the CVO, and the underlying PO packages during phase 0. This can be done by reconciling an "aggregate" ClusterOperator that reflects the overall state of installing PO packages. Future phases may require PO packages to reconcile OLM's OperatorCondition API, or the directly manage a ClusterOperator resource to express their health and overall state.
+
+In order to address the existing plain provisioner limitation, this proposal also recommends expanding the plain+v0 bundle format, and enabling users to define health checks through a dedicated metadata file. A valid metadata file may be named "bundle.yaml" so clients can know where to find this file.
+
+For phase 0, the registry+v1 provisioner would be responsible for automatically generating this metadata file for PO packages. Future phases may introduce support for defining PO packages using OLM's plain bundle format, which will require operator authors to explicitly define this metadata file themselves.
+
+These metadata-level health checks can be expressed using JSONPath templates to declaratively define custom probe logic. When the plain provisioner is processing unpacked Bundle contents, and a manifests file has been defined, it’s responsible for serializing the values defined in the metadata file, and evaluating any defined custom probes. When an an individual probe evaluation fails, that state will be reflected in the BundleDeployment’s status.
+
+In the case that a probe was misconfigured, an admin may elect to toggle this behavior off entirely. The ProvisionerClass API, which doesn’t exist today, would be a candidate resource for providing this escape hatch behavior. Without that API in place today, the `plain.core.rukpak.io/enable-probes` BundleDeployment annotation key may serve as a short term solution. The plain provisioner implementation would be responsible for respecting that annotation key.
```

Comment:
> The ProvisionerClass API, which doesn’t exist today, would be a candidate resource for providing this escape hatch behavior.

Assuming that api is intended to be a cluster scoped configuration api for the provisioner, i think we'd want the escape hatch to be on the BundleDeployment or whatever resource requested the operator install, not a global setting for the provisioner in general.  (So i see the annotation as being more than a short-term solution)

Reply:
Marking this as resolved as we've moved rukpak health checks to out-of-scope for this phase.